hibp/table_password.go

```diff
@@ -2,125 +2,70 @@ package hibp
 
 import (
 	"context"
+	"crypto/sha1"
+	"fmt"
 
-	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
-	"github.com/turbot/steampipe-plugin-sdk/plugin"
-	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
-	"gitlab.com/wedtm/go-hibp"
+	"github.com/turbot/steampipe-plugin-sdk/v3/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin"
 )
 
+type psswrdRow struct {
+	Plain *string
+	Hash  *string
+	Count *int64
+}
+
 func tablePassword() *plugin.Table {
 	return &plugin.Table{
 		Name:        "hibp_password",
 		Description: "Password (hashes) tracked by HIBP",
 		List: &plugin.ListConfig{
-			KeyColumns: plugin.AnyColumn([]string{"prefix", "hash"}),
+			KeyColumns: plugin.AnyColumn([]string{"plain", "hash"}),
 			Hydrate:    listPasswords,
 		},
-		Get: &plugin.GetConfig{
-			KeyColumns: plugin.SingleColumn("hash"),
-			Hydrate:    getPassword,
-		},
 		Columns: []*plugin.Column{
-			{Name: "prefix", Type: proto.ColumnType_STRING, Description: "The first five characters of the hash", Transform: transform.From(transformPrefix), Hydrate: hydratePrefix},
-			{Name: "hash", Type: proto.ColumnType_STRING, Description: "The hash of the compromised password.", Transform: transform.From(transformHash), Hydrate: hydrateHash},
+			{Name: "plain", Type: proto.ColumnType_STRING, Description: "The plain-text of the compromised password (sent as a hash to the API)."},
```

Comment:
How does this change affect usage of the table? Am I still able to pass a prefix anywhere of 5 chars to search a range, like the API allows?

Reply:
The API allows it, but the SDK does not expose all the results. I have opened up an issue at https://github.com/wneessen/go-hibp/issues/14 requesting for it.